singularity-eos/closure/mixed_cell_models.hpp

```diff
@@ -462,22 +462,87 @@ pte_closure_flag_with_line_impl(EOSIndexer &&eoss, const Real Volume, const Real
 // RealIndexer types may be different because some might be arrays and
 // some might be pointers.
 template <int nmat, typename T1, typename T2, typename T3>
-PORTABLE_INLINE_FUNCTION static void pte_residual(const Real utot, T1 &&vfrac, Real *u,
-                                                  T2 &&temp, T3 &&press, Real *residual) {
+PORTABLE_INLINE_FUNCTION static void pte_residual(const Real vfrac_tot, const Real utot,
+                                                  T1 &&vfrac, Real *u, T2 &&temp,
+                                                  T3 &&press, Real *residual) {
   Real vsum = 0.0;
   Real esum = 0.0;
   for (int m = 0; m < nmat; ++m) {
     vsum += vfrac[m];
     esum += u[m];
   }
-  residual[0] = 1.0 - vsum;
+  residual[0] = vfrac_tot - vsum;
```

Comment:
I think that `vfrac_tot` will never _not_ be 1.0 in `pte_closure_josh_impl`, which is the only place this is used. So I don't think we were ever bitten by this bug, but it could bite in the future if somebody isn't reading the code carefully.